### PR TITLE
BE-1389 Fixing fields

### PR DESCRIPTION
**Pull request by josecordaz**
https://pinpt-hq.atlassian.net/browse/BE-1389

**Description:**

- Added missing fields

### CODE REVIEW

File: pkg/git/branches.go [hunk 1]
```diff
@@ -12,60 +12,54 @@ import (
...
+	RepoID string `json:"repo_id" bson:"repo_id" yaml:"repo_id" faker:"-"`
 }
 
-func newBranchFromRipsrc(b ripsrc.Branch) (res Branch) {
+func newBranchFromRipsrc(b ripsrc.Branch, repoid string) (res Branch) {
 	res.ID = b.ID
+	res.RefID = ""
+	res.RefType = ""
+	res.CustomerID = ""
+	res.Hashcode = ""
```
Comment (jhaynie) [inline pkg/git/branches.go]: need to set CustomerID to the customer.  we should have that in the agent, can we pass it in? the RefType should be the system such as "github", "bitbucket", etc.

File: pkg/git/commitactivity.go [hunk 2]
```diff
@@ -21,6 +24,16 @@ type authorSum struct {
...
+	// AuthorRefID string `json:"author_ref_id" bson:"author_ref_id" yaml:"author_ref_id" faker:"-"`
+	// // Date the change data in RFC3339 format
+	// Date string `json:"date" bson:"date" yaml:"date" faker:"-"`
+	// // Comment if the line is a comment
+	// Comment bool `json:"comment" bson:"comment" yaml:"comment" faker:"-"`
+	// // Code if the line is sourcecode
+	// Code bool `json:"code" bson:"code" yaml:"code" faker:"-"`
+	// // Blank if the line is a blank line
+	// Blank bool `json:"blank" bson:"blank" yaml:"blank" faker:"-"`
 }
```
Comment (jhaynie) [inline pkg/git/commitactivity.go]: remove commented code

File: pkg/git/commitactivity.go [hunk 3]
```diff
@@ -84,17 +97,33 @@ func (s *loggerForRipsrc) Debug(msg string, args ...interface{}) {
...
+		return 1
+	case "removed":
+		return 2
+	}
+	return -1
+}
+
 // GenerateCommitActivity will generate a commit activity
 // If no new commits, lastProcessedSHA == ""
 func GenerateCommitActivity(ctx context.Context, logger log.Logger, inName string, dirs gitDirs, name string, repoid string, branch string, refType string, fromsha string, committerExcludePattern *regexp.Regexp, commitcount int) (lastProcessedSHA string, _ error) {
```
Comment (jhaynie) [inline pkg/git/commitactivity.go]: rename to GenerateBlame

File: pkg/git/commitactivity.go [hunk 4]
```diff
@@ -109,13 +138,15 @@ func GenerateCommitActivity(ctx context.Context, logger log.Logger, inName strin
...
 		var detail *commitDetail
-		var additions, deletions int
+		var additions, deletions int64
 		var excludedFilesCount int64
 		var lastBlame *ripsrc.BlameResult
 
 		for commit := range results {
 			lastProcessedSHA = commit.SHA
 
+			fmt.Println("Processing SHA", commit.SHA)
```
Comment (jhaynie) [inline pkg/git/commitactivity.go]: remove

File: pkg/git/commitactivity.go [hunk 5]
```diff
@@ -173,41 +214,47 @@ func GenerateCommitActivity(ctx context.Context, logger log.Logger, inName strin
...
-				additions += cf.Additions
-				deletions += cf.Deletions
+				additions += int64(cf.Additions)
+				deletions += int64(cf.Deletions)
 				var lic string
 				if blame.License != nil {
 					lic = blame.License.Name
 				}
 				files = append(files, commitDetailFile{
+					CommitID:          commit.SHA,
```
Comment (jhaynie) [inline pkg/git/commitactivity.go]: maybe rename commitFile so we don't confuse with the old commitDetail

File: pkg/git/commitactivity.go [hunk 6]
```diff
@@ -144,13 +175,23 @@ func GenerateCommitActivity(ctx context.Context, logger log.Logger, inName strin
 				// first we need to calculate the per author stats since the
 				// commit activity is per commit per blame author
 				authorSums := make(map[string]*authorSum)
+				authorLines := make(map[string][]BlameLines)
```
Comment (jhaynie) [inline pkg/git/commitactivity.go]: this isn't correct. you're adding these by author which will make the blame lines out of order.

The blame lines are each line of code in the source file. if you put in a map by user_id, we won't have any idea what line it is (it's a hash map).

i.e. the array is line 0, line 1, line 2 (each line in the file) and the change that was made, by who, what the change is, etc.

we don't need it to by by user.  it should be ALL users on the Blame object per change (regardless of who made the change).

the point of this object is to basically reflect the most recent change to each file in the commit (as of the new change).

File: pkg/git/commitactivity.go [hunk 7]
```diff
@@ -144,13 +175,23 @@ func GenerateCommitActivity(ctx context.Context, logger log.Logger, inName strin
...
 						authorSums[userid] = entry
+						authorLines[userid] = make([]BlameLines, 0)
+					} else {
+						authorLines[userid] = append(authorLines[userid], BlameLines{
+							AuthorRefID: line.Email,
+							Date:        line.Date.Format("2006-01-02T15:04:05.000000Z-07:00"),
+							Comment:     line.Comment,
+							Code:        line.Code,
+							Blank:       line.Blank,
+						})
```
Comment (jhaynie) [inline pkg/git/commitactivity.go]: see above but just use BlameLines as it in the order that it comes in the array

File: pkg/git/commitactivity.go [hunk 8]
```diff
@@ -173,41 +214,47 @@ func GenerateCommitActivity(ctx context.Context, logger log.Logger, inName strin
...
+					CommitID:          commit.SHA,
+					ID:                commit.SHA,
+					RefType:           refType,
+					RefID:             commit.SHA,
+					CustomerID:        customerID,
+					Hashcode:          "",
+					RepoID:            repoid,
+					Status:            string(cf.Status),
+					Ordinal:           0,
+					CreatedAt:         datetime.TimeToEpoch(blame.Commit.Date),
```
Comment (jhaynie) [inline pkg/git/commitactivity.go]: ignore hashcode (empty is fine)

for ordinal, use the ordinal logic we had before.  if that wasn't there (can't remember), just do this:


at the top of the file before the loop:

```
ordinal := datetime.EpochNow()
```

then each iteration through the loop just increment the ordinal

```
ordinal++
```

the ordinal is just meant to represent a monotonically increasing number (as long as older commits are smaller than newer commits)

File: pkg/git/commitactivity.go [hunk 9]
```diff
@@ -173,41 +214,47 @@ func GenerateCommitActivity(ctx context.Context, logger log.Logger, inName strin
...
+				additions += int64(cf.Additions)
+				deletions += int64(cf.Deletions)
 				var lic string
 				if blame.License != nil {
 					lic = blame.License.Name
 				}
 				files = append(files, commitDetailFile{
+					CommitID:          commit.SHA,
+					ID:                commit.SHA,
+					RefType:           refType,
```
Comment (jhaynie) [inline pkg/git/commitactivity.go]: id needs to be a hashed ID, not the sha:

https://github.com/pinpt/go-datamodel/blob/master/sourcecode/commit_file.go#L266

File: pkg/git/commitactivity.go [hunk 10]
```diff
@@ -221,41 +268,63 @@ func GenerateCommitActivity(ctx context.Context, logger log.Logger, inName strin
...
+							Blanks:         0,
+							Comments:       0,
+							Filename:       blame.Filename,
+							Language:       blame.Language,
+							ID:             hash.Values(blame.Filename, blame.Commit.SHA, branch, authorID),
+							Sha:            blame.Commit.SHA,
+							DateAt:         datetime.TimeToEpoch(blame.Commit.Date),
+							Date:           blame.Commit.Date.Format("2006-01-02T15:04:05.000000Z-07:00"),
+							RepoID:         repoid,
+							Lines:          make([]BlameLines, 0),
```
Comment (jhaynie) [inline pkg/git/commitactivity.go]: as mentioned above, this can't happen now. blame lines will always be >0

File: pkg/git/commitactivity.go [hunk 11]
```diff
@@ -264,22 +333,33 @@ func GenerateCommitActivity(ctx context.Context, logger log.Logger, inName strin
...
-					Blanks:       commitBlanks,
-					Size:         commitSize,
-					Complexity:   commitComplexity,
+					Deletions:      deletions,
+					Additions:      additions,
+					ID:             commit.SHA,
+					Sha:            commit.SHA,
+					Message:        lastBlame.Commit.Message,
+					URL:            "",
+					AuthorRefID:    "",
```
Comment (jhaynie) [inline pkg/git/commitactivity.go]: need commit URL from the system (maybe have to pass it in from the integration)

leave author and committer id empty until Robin finishes his change and we can use his logic

File: pkg/git/commitactivity.go [hunk 12]
```diff
@@ -264,22 +333,33 @@ func GenerateCommitActivity(ctx context.Context, logger log.Logger, inName strin
...
-					Branch:       strings.ToLower(branch),
-					Loc:          commitLoc,
-					Sloc:         commitSloc,
-					Comments:     commitComments,
-					Blanks:       commitBlanks,
-					Size:         commitSize,
-					Complexity:   commitComplexity,
+					Deletions:      deletions,
+					Additions:      additions,
+					ID:             commit.SHA,
```
Comment (jhaynie) [inline pkg/git/commitactivity.go]: ID needs to be hashed value:

https://github.com/pinpt/go-datamodel/blob/master/sourcecode/commit.go#L258

RefID should be the SHA

File: pkg/git/commitactivity.go [hunk 13]
```diff
@@ -264,22 +333,33 @@ func GenerateCommitActivity(ctx context.Context, logger log.Logger, inName strin
...
+					RepoID:         repoid,
+					Files:          files,
+					FilesChanged:   int64(len(files)),
+					Ordinal:        lastBlame.Commit.Ordinal,
+					GpgSigned:      lastBlame.Commit.Signed,
+					Excluded:       int64(len(files)) == excludedFilesCount && excludedFilesCount != 0,
+					CreatedAt:      datetime.TimeToEpoch(lastBlame.Commit.Date),
+					Created:        lastBlame.Commit.Date.Format("2006-01-02T15:04:05.000000Z-07:00"),
+					Branch:         strings.ToLower(branch),
+					Loc:            commitLoc,
```
Comment (jhaynie) [inline pkg/git/commitactivity.go]: not sure we should lower case the branch

File: pkg/git/branches.go [hunk 14]
```diff
@@ -12,60 +12,54 @@ import (
 
 // Branch is a copy of ripsrc.Branch with json tags set
```
Comment (xzandrew) [inline pkg/git/branches.go]: Should prob. remove this comment. It's no longer a simple copy of ripsrc.Branch.

**Pull request merged**